orttraining/orttraining/training_api/optimizer.h

```diff
@@ -86,24 +138,30 @@ struct Optimizer {
   }
 
  private:
+  void Initialize(const std::string& optim_path_or_bytes,
+                  const onnxruntime::SessionOptions& session_options,
+                  const Environment& env,
+                  const std::vector<std::shared_ptr<IExecutionProvider>>& providers);
+
   int64_t GetStep() const {
     return optimizer_state_->step;
   }
 
-  // Generates optimizer momentum states for applicable optimizer types
-  Status GenerateMomentumNamedStates();
+  // Generates optimizer momentum states for parameters that require grad.
+  Status GenerateMomentumNamedStates(OptimizerCheckpointState& optimizer_checkpoint_states);
   // Constructs the ortvalue inputs to be fed to the graph
-  // at each step
+  // at each step.
   Status ConstructInputs();
 
-  // TODO: load this info from checkpoint
-  OptimizerType optimizer_type_ = OptimizerType::AdamW;
+  std::shared_ptr<OptimizerAlgorithmBase> optimizer_algo_shared_ptr_;
```

Comment:
Can we make this unique pointer and make optimizer own this? Does it have to be a shared pointer?

Reply:
Sure fixed it. Thanks!